fe/fe-core/src/main/java/org/apache/doris/nereids/stats/StatsCalculator.java

```diff
@@ -70,14 +72,23 @@ public StatsCalculator(GroupExpression groupExpression) {
      * Do estimate.
      */
     public void estimate() {
+
         StatsDeriveResult stats = groupExpression.getPlan().accept(this, null);
         groupExpression.getOwnerGroup().setStatistics(stats);
-        Plan plan = groupExpression.getPlan();
-        long limit = plan.getLimit();
-        if (limit != -1) {
-            stats.setRowCount(Math.min(limit, stats.getRowCount()));
-        }
         groupExpression.setStatDerived(true);
+
+    }
+
+    @Override
+    public StatsDeriveResult visitLogicalLimit(LogicalLimit<Plan> limit, Void context) {
+        StatsDeriveResult stats = groupExpression.getCopyOfChildStats(0);
+        return stats.updateRowCountByLimit(limit.getLimit() + limit.getOffset());
+    }
+
+    @Override
+    public StatsDeriveResult visitPhysicalLimit(PhysicalLimit<Plan> limit, Void context) {
+        StatsDeriveResult stats = groupExpression.getCopyOfChildStats(0);
+        return stats.updateRowCountByLimit(limit.getLimit() + limit.getOffset());
```

Comment:
i think we should set row count to limit.getLimit() directly, @Kikyou1997  PTAL

Reply:
I think so